### PR TITLE
Recipe updates for non-amd64 architectures 

### PR DESCRIPTION
**Pull request by aandvalenzuela**
This PR includes `aarch64` and `pcc64le` archs in the recipes. I have kept compiler definition for `amd64`, but it needs to be modified for testing to the corresponding compiler. Aside, this PR also removes padding in the installation path.

### CODE REVIEW

**Comment by iarspider** [comment]
@aandvalenzuela could you please check if packages in `backports` are already in 0.19?

File: repos/cms/packages/oracle-fake/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
```
Comment (iarspider) [inline repos/cms/packages/oracle-fake/package.py]: Please remove the boilerplate

File: repos/cms/packages/tkonlinesw-fake/package.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
```
Comment (iarspider) [inline repos/cms/packages/tkonlinesw-fake/package.py]: Same here

**Review by iarspider — CHANGES_REQUESTED** [review]

**Comment by aandvalenzuela** [comment]
@iarspider is there any reason why branch `CMSSW_12_6_X` does not share the git history with branch `CMSSW_12_5_pre4`?

**Comment by iarspider** [comment]
I wanted to keep history clean

**Comment by aandvalenzuela** [comment]
> @aandvalenzuela could you please check if packages in `backports` are already in 0.19?

They are not, I'd say: https://github.com/iarspider/cms-spack-repo/tree/CMSSW_13_0_X/repos/backports/packages
Backported packages are:
-  `benchmark` (upstream: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/benchmark/package.py), 
- `igprof` (https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/igprof/package.py), 
- `professor` (https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/professor/package.py),
-  `rapidjson`(https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/rapidjson/package.py).

**Review by iarspider — APPROVED** [review]

**Comment by aandvalenzuela** [comment]
Many thanks @iarspider!

**Comment by iarspider** [comment]
Can you try porting this to 13_0_X branch? I want to setup an IB using that branch before X-mas

**Comment by aandvalenzuela** [comment]
Yes, I will try. I think it can take some time due to uncommon histories. `CMSSW_13_0_X` doesn't build for me, is it for you?

**Comment by iarspider** [comment]
Everything except cmssw should build

**Comment by iarspider** [comment]
cmssw is outdated, you can add any recent (e.g. today's) tag in there. Just keep the placeholder line.

**Comment by iarspider** [comment]
> Yes, I will try. I think it can take some time due to uncommon histories. `CMSSW_13_0_X` doesn't build for me, is it for you?

You can try getting this PR [as a diff](https://patch-diff.githubusercontent.com/raw/iarspider/cms-spack-repo/pull/34.diff), then figure out any patch conflicts you get.

**Pull request merged**